### DISCUSSION TITLE
How to use ffmpeg in a linux docker container

### DISCUSSION THREAD

**Discussion by byMuchi** [Tyrrrz/YoutubeExplode#669] [Q&A] [answered]
Hello,

i tried building a Web Application using this library for downloading mp3 files.

When i execute the app in visual studio i can download and convert the video without problems.
Now i tried putting the application on a webserver running docker in my network, so i integrated docker support in my app.

When i tested everything, i got an error at the "DownloadAsync"-Part which says
"FFmpeg exited with a non-zero exit code (255)."
So i think that the ffmpeg conversion is the main problem here.

Is it possible to get this to work in a linux docker container or is this just impossible?
I didnt really find anything usefull on Google so i thought i might find some help here.

Thanks in advance & kind regards
Cedric

**Comment by Tyrrrz** [comment, marked as answer]
It should be possible. What does your dockerfile look like?

**Reply by byMuchi** [reply]
My dockerfile is just the default filethat is generated when  i enabled docker support in visual studio for the project, but here it is:

```
FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["youtubewebdownloader/youtubewebdownloader.csproj", "youtubewebdownloader/"]
RUN dotnet restore "youtubewebdownloader/youtubewebdownloader.csproj"
COPY . .
WORKDIR "/src/youtubewebdownloader"
RUN dotnet build "youtubewebdownloader.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "youtubewebdownloader.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "youtubewebdownloader.dll"]
```

My ffmpeg.exe file i located in the wwwroot folder in another folder called "ffmpeg"
and my DownloadAsync-Call looks like this
```
await youtube.Videos.DownloadAsync(videoUrl, savePath, o => o
.SetContainer("mp3")
.SetPreset(ConversionPreset.UltraFast)
.SetFFmpegPath(Path.Combine(".","wwwroot", "ffmpeg", "ffmpeg.exe"))
);
```

**Reply by Tyrrrz** [reply]
Well if you're running on Linux, it should be a Linux ffmpeg binary, so not `ffmpeg.exe`. You can find various download options here: https://github.com/vot/ffbinaries-prebuilt/releases

You can also use something like this to download the right binary dynamically: https://github.com/Tyrrrz/YoutubeExplode/blob/ee828534c098f47f83a81a6601ba99b4caf98ed8/YoutubeExplode.Converter.Tests/Fixtures/FFmpegFixture.cs

Alternatively, just use `apt-get` or something similar in your dockerfile and install ffmpeg on path.

**Reply by byMuchi** [reply]
Okay thank you.

I tried the first approach and it works now, so for now i stay with it, but i might try the second suggestion at one point.
